### PR TITLE
Initial Deploy

### PR DESCRIPTION
**Pull request by tinaliu0-1629521**

### CODE REVIEW

File: CloudFormationLab.yml [hunk 1]
```diff
@@ -34,9 +33,8 @@
         Type: 'AWS::EC2::Subnet'
         Properties:
           VpcId: !Ref PubPrivateVPC
-          AvailabilityZone: us-east-1a
           CidrBlock: 172.31.3.0/24
-          MapPublicIpOnLaunch: false
+          MapPublicIpOnLaunch: false #makes it a private subnet
```
Comment (sidewayset) [inline CloudFormationLab.yml]: Does not make it private subnet, just does not give public  on launch by default. You can still get public ip if you want

File: CloudFormationLab.yml [hunk 2]
```diff
@@ -47,20 +45,20 @@
...
-      #   Properties:
-      #     Tags:
-      #       - Key: Name
-      #         Value: !Join [_, [!Ref 'AWS::StackName']]
-      #       - Key: Network
-      #         Value: Public
+      InternetGateway:
+        Type: 'AWS::EC2::InternetGateway' #creates gateway
+        Properties:
+          Tags:
```
Comment (sidewayset) [inline CloudFormationLab.yml]: Recognize that tags are not required and are used to track resources

File: CloudFormationLab.yml [hunk 3]
```diff
@@ -90,36 +88,35 @@
...
-      #   Type: "AWS::EC2::NatGateway"
-      #   DependsOn: NatPublicIP
-      #   Properties: 
-      #     AllocationId: !GetAtt NatPublicIP.AllocationId
-      #     SubnetId: !Ref PublicSubnet1
+      NatGateway:
+        Type: "AWS::EC2::NatGateway" #allows private subnet to connect w/ internet or other AWS services
+        Properties: 
+          AllocationId: !GetAtt NatPublicIP.AllocationId #EIP to associate w/ NAT gateway
+          SubnetId: !Ref PublicSubnet1
```
Comment (sidewayset) [inline CloudFormationLab.yml]: Nobody asked for Nat Gateway ;)

File: CloudFormationLab.yml [hunk 4]
```diff
@@ -90,36 +88,35 @@
...
-      # NatPublicIP:
-      #   Type: "AWS::EC2::EIP"
-      #   DependsOn: PubPrivateVPC
-      #   Properties:
-      #     Domain: vpc
+      NatPublicIP:
+        Type: "AWS::EC2::EIP"
+        DependsOn: PubPrivateVPC
+        Properties:
+          Domain: vpc
```
Comment (sidewayset) [inline CloudFormationLab.yml]: No NAT

File: CloudFormationLab.yml [hunk 5]
```diff
@@ -90,36 +88,35 @@
...
             - Key: Network
               Value: Private
       
       PrivateRoute:
-        Type: 'AWS::EC2::Route'
+        Type: 'AWS::EC2::Route' #specifies route in route table 
         Properties:
           RouteTableId: !Ref PrivateRouteTable
           DestinationCidrBlock: 0.0.0.0/0
           NatGatewayId: !Ref NatGateway
```
Comment (sidewayset) [inline CloudFormationLab.yml]: No Nat

**Review by sidewayset — CHANGES_REQUESTED** [review]
Remove commented out code

File: CloudFormationLab.yml (hunk 2 above)
Comment (tinaliu0-1629521) [reply CloudFormationLab.yml]: Yeah! You would prefer I took them out?

Comment (sidewayset) [reply CloudFormationLab.yml]: @tinaliu0-1629521 up to you, just pointing it out

File: CloudFormationLab.yml (hunk 1 above)
Comment (tinaliu0-1629521) [reply CloudFormationLab.yml]: So... need to find way to make public IP unavailable?

Comment (sidewayset) [reply CloudFormationLab.yml]: no, false is correct here but comment is wrong. It does not make subnet private. Associating this subnet with routing table without internate gateway is what makes it private subnet. This simply says do not assign public facing IP

**Review by sidewayset — APPROVED** [review]

**Pull request merged**